text/src/autogluon/text/automm/models/fusion.py

```diff
@@ -207,3 +208,172 @@ def get_layer_ids(self,):
             assert n in name_to_id
 
         return name_to_id
+
+
+class MultimodalFusionTransformer(nn.Module):
+    """
+    Use MLP to fuse different models' features (single-modal and multimodal).
+    Specifically, it adapts the features of each model to specified dimensions,
+    concatenates the adapted features, and fuses the features through MLP.
+    """
+
+    def __init__(
+            self,
+            prefix: str,
+            models: list,
+            hidden_features: List[int],
+            heads: List[int],
+            num_classes: int,
+            adapt_in_features: Optional[str] = None,
+            dropout_prob: Optional[float] = 0.5,
+            loss_weight: Optional[float] = None,
+    ):
+        super().__init__()
+        if loss_weight is not None:
+            assert loss_weight > 0
+
+        self.loss_weight = loss_weight
+        self.model = nn.ModuleList(models)
+
+        raw_in_features = [per_model.out_features for per_model in models]
+        if adapt_in_features is not None:
+            if adapt_in_features == "min":
+                base_in_feat = min(raw_in_features)
+            elif adapt_in_features == "max":
+                base_in_feat = max(raw_in_features)
+            else:
+                raise ValueError(f"unknown adapt_in_features: {adapt_in_features}")
+
+            self.adapter = nn.ModuleList(
+                [nn.Linear(in_feat, base_in_feat) for in_feat in raw_in_features]
+            )
+
+            # in_features = base_in_feat * len(raw_in_features)
+            in_features = base_in_feat
+        else:
+            self.adapter = nn.ModuleList(
+                [nn.Identity() for _ in range(len(raw_in_features))]
+            )
+            in_features = sum(raw_in_features)
```

Comment:
This is not correct. You may design it in a way that in_features are pre-defined.

Reply:
I removed this part. `adapt_in_features` is necessary for `fusion_transformer`, as the features from different model are concated at `dim=1`. Thus, the absences of adapter may not be feasible.